src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROW_FACTOR = 1.5;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        ensureCapacity();
+        elementData[size] = value;
+        size++;
+    }
+
+    private void ensureCapacity() {
+
+        if (size == elementData.length) {
+            grow();
+        }
+    }
 
+    private void grow() {
+        int newCapacity = (int) (elementData.length * GROW_FACTOR);
+        T[] newArray = (T[]) new Object[newCapacity];
+        System.arraycopy(elementData, 0, newArray, 0, size);
+        elementData = newArray;
```

Comment:
The `grow` method should not contain conditional logic for resizing the array. It's better to move the condition check outside this method to adhere to the single responsibility principle and ensure the method name reflects its action correctly.